### PR TITLE
add tests for 2 permissions api

### PR DESCRIPTION
**Pull request by qnkhuat**
Tests for these 2 APIs
- API `POST /permissions/group`
- API `DELETE /permission/group/:group-id`

Part of #28598

### CODE REVIEW

File: enterprise/backend/test/metabase_enterprise/advanced_permissions/api/group_manager_test.clj [hunk 1]
```diff
@@ -28,26 +28,44 @@
               (update-group [user status group]
                 (testing (format ", update group with %s user" (mt/user-descriptor user))
                   (let [new-name (mt/random-name)]
-                    (mt/user-http-request user :put status (format "permissions/group/%d" (:id group)) {:name new-name}))))]
+                    (mt/user-http-request user :put status (format "permissions/group/%d" (:id group)) {:name new-name}))))
+
+              (delete-group [user status group-manager?]
+                (testing (format ", delete group with %s user" (mt/user-descriptor user))
+                  (let [user-id (u/the-id (if (keyword? user) (mt/fetch-user user) user))]
```
Comment (tsmacdonald) [inline enterprise/backend/test/metabase_enterprise/advanced_permissions/api/group_manager_test.clj:35]: When would `user` not be a keyword?

File: test/metabase/api/permissions_test.clj [hunk 2]
```diff
@@ -102,6 +104,33 @@
...
+   (testing "group name is required"
+     (is (= {:errors {:name "value must be a non-blank string."}}
+            (mt/user-http-request :crowberto :post 400 "permissions/group" {:name nil}))))))
+
+(deftest delete-group-test
+  (testing "DELETE /permissions/group/:id"
+    (testing "happy path"
+      (t2.with-temp/with-temp [PermissionsGroup {group-id :id} {:name "Test group"}]
+        (mt/user-http-request :crowberto :delete 204 (format "permissions/group/%d" group-id))
+        (is (= 0 (t2/count PermissionsGroup :name "Test group")))))
```
Comment (tsmacdonald) [inline test/metabase/api/permissions_test.clj:127]: Maybe I'm hyper, but I usually put an

```
(is (= 1 (t2/count PermissionsGroup :name "Test group"))
```
beforehand just to be absolutely sure that the test does what I think it does. Consider, for example, that `(t2/count PermissionsGroup :name "Test Group")` (with a typo: capital `G`) will always return 0.

**Review by tsmacdonald — APPROVED** [review]
:pray:

File: enterprise/backend/test/metabase_enterprise/advanced_permissions/api/group_manager_test.clj (hunk 1 above)
Comment (qnkhuat) [reply enterprise/backend/test/metabase_enterprise/advanced_permissions/api/group_manager_test.clj:35]: lots, here is an example: https://github.com/metabase/metabase/blob/f730f6afba3713de486b75d0fea455c0b3a1e725/enterprise/backend/test/metabase_enterprise/advanced_permissions/api/group_manager_test.clj#L52

File: test/metabase/api/permissions_test.clj (hunk 2 above)
Comment (qnkhuat) [reply test/metabase/api/permissions_test.clj:127]: yea I do that as well but in this case the test is quite simple and the with-temp is right above it so I think it's fine.

**Pull request merged**